radar_control/radar_control.py

```diff
@@ -686,24 +688,26 @@ def radar():
                                          sm.COLOR("blue", round(bound_time_remaining, 6)))
                         rad_ctrl_print(msg)
 
-                        if bound_time_remaining < aveperiod.intt*1e-3:
+                        if bound_time_remaining < aveperiod.intt * 1e-3:
                             # reduce the integration period to only the time remaining
                             # until the next scan boundary.
+                            # TODO: Check for integration_period_done_time < 0
```

Comment:
I think it's actually bound_time_remaining that we want to check, to be sure it's not < 0 . integration_period_done_time is a datetime.datetime; bound_time_remaining is a timedelta
```suggestion
                            # TODO: Check for bound_time_remaining > 0
                            # to be sure there is actually time to run this intt
                            # (if bound_time_remaining < 0, we need a solution to 
                            # reset)
```